stackrox/stackrox
# ROX-13888: Migrate permission sets referencing Policy and VulnerabilityReports resources

## Description

This is the second and last PR in a series of PRs to remove the resources Policiy and VulnerabilityReports in favor of the WorkflowAdministration resource.

The PR itself covers the migration that touches all permission sets that reference the resource.

The migration follows the same logic as previously for resource removals: the lowest access level will be chosen of either the resource that should be removed or the replacing resource.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

- see CI passing.

Manual tests:
1. Add a test permission set that references both resources that will be replaced with this migration:
```bash
roxcurl /v1/permissionsets -X POST -d '{"id":"","name":"test-perm-set","description":"testing migration","resourceToAccess":{"Policy":"READ_ACCESS","VulnerabilityReports":"READ_WRITE_ACCESS"}}'
```

2. Update to the central image within this PR.

3. Verify within the logs that the migration has run:
```log
Migrator: 2023/06/07 14:25:37.853213 log.go:18: Info: Found DB at version 184, which is less than what we expect (185). Running migrations...
Migrator: 2023/06/07 14:25:37.864948 log.go:18: Info: Successfully updated DB from version 184 to 185
```

4. Verify within the API that the permission set is the expected one:
```bash
        {
            "id": "c8e24b11-a23c-44e4-8b74-83dc2131fb35",
            "name": "test-perm-set",
            "description": "testing migration",
            "resourceToAccess": {
                "WorkflowAdministration": "READ_ACCESS"
            },
            "traits": null
        }
```
